tests/test_strings.py

```diff
@@ -532,5 +533,15 @@ def test_atbash_cipher(self):
         self.assertEqual("AttaCK at DawN", atbash("ZggzXP zg WzdM"))
         self.assertEqual("ZggzXP zg WzdM", atbash("AttaCK at DawN"))
         
+class TestKnuthMorrisPratt(unittest.TestCase):
+    """[summary]
+    Test for the file knuth_morris_pratt.py
+
+    Arguments:
+        unittest {[type]} -- [description]
+    """
+    def test_knuth_morris_pratt(self):
+        self.assertEqual([0, 4], knuth_morris_pratt("abcdabc", "abc"))
+
```

Comment:
think more about possible edge cases. what if the string does not match? what if more than 1 possible matches?

Reply:
Sorry, the description of the algorithm was quite unkind. And I edited it now. If the string does not match, this code returns an empty list. If there are more than 1 cases, list of the start index will be returned.

Reply:
Sorry if I wasn't clear. I meant unit tests should cover those edge cases.

Reply:
I added test case for those cases. Can you check it?

Reply:
thanks!